packages/keyring-snap-bridge/src/SnapKeyring.ts

```diff
@@ -1497,6 +1527,44 @@ export class SnapKeyring {
     );
   }
 
+  /**
+   * Construct a map of selected accounts by Snap ID and update the in-memory selected accounts map.
+   *
+   * @param accounts - The accounts to construct the map from.
+   * @returns A map of selected accounts by Snap ID.
+   */
+  #constructAndUpdateSelectedAccountsMap(
+    accounts: AccountId[],
+  ): Record<SnapId, AccountId[]> {
+    const map = accounts.reduce<Record<SnapId, AccountId[]>>((acc, account) => {
+      const snapId = this.#accounts.getSnapId(account);
+      if (!snapId) {
+        return acc;
+      }
+      acc[snapId] = acc[snapId] ?? [];
+      acc[snapId].push(account);
+      return acc;
+    }, {});
+    this.#selectedAccounts = map;
+    return map;
+  }
+
+  /**
+   * Set the selected accounts.
+   *
+   * @param accounts - The accounts to set as selected.
+   */
+  async setSelectedAccounts(accounts: AccountId[]): Promise<void> {
+    const accountsMap = this.#constructAndUpdateSelectedAccountsMap(accounts);
+    await Promise.allSettled(
+      Object.entries(accountsMap).map(([snapId, accountIds]) =>
+        this.#snapClient
+          .withSnapId(snapId as SnapId)
+          .setSelectedAccounts(accountIds),
+      ),
+    );
```

Comment:
We should also log the errors in case something went wrong, we can use `console.warn('SnapKeyring - ...');` like for the other kind of non-critical errors we have